core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -218,6 +220,41 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
     }
   }
 
+  /**
+   * INTERNAL API
+   */
+  @InternalApi private[akka] def rejectPubSubTooFarAhead[Event](
+      maxAheadOfBacktracking: JDuration): Flow[EventEnvelope[Event], EventEnvelope[Event], NotUsed] = {
+    Flow[EventEnvelope[Event]]
+      .statefulMapConcat(() => {
+        // track backtracking offset
+        var latestBacktracking = Instant.EPOCH
+        env => {
+          env.offset match {
+            case t: TimestampOffset =>
+              if (env.eventOption.isEmpty) {
```

Comment:
for gRPC RES transport, empty event is also events not originating in the remote DC but replicated there, so this may not hold to detect backtracking for that use case?

Reply:
um, then I have to look into that. we might have more places where empty is the way to indicate that it's from backtracking, such as the failing check in the offset store

Reply:
This is what I do on the producing side: https://github.com/akka/akka-projection/pull/757/files#diff-7c1a8b28b12936014dbcd01080bb2d9469c6a3de48b13309cc45f97b71414e76R93

Reply:
So the logic in `R2dbcProjectionImpl#loadEnvelope` will at least anyway load each of those payloads it seems, I had missed that before.

Reply:
I wonder if I should just revert/replace that optimization with a specific payload that is not filtered instead in the gRPC transport.